x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeFormatProcessor.java

```diff
@@ -87,6 +148,28 @@ public Object format(Object timestamp, Object pattern, ZoneId zoneId) {
                 );
             }
         }
+
+        private String formatUsingMappings(TemporalAccessor temporalAccessor, String pattern) {
+            for (DateTimeFormatMapping mapping : JAVA_TIME_TO_CHAR_MAPPINGS) {
+                if (pattern.contains(mapping.getPattern())) {
+                    var middle = mapping
+                        .getAdditionalMapper()
+                        .apply(format(mapping.getJavaPattern(), temporalAccessor));
+
+                    var parts = pattern.split(mapping.getPattern(), 2);
```

Comment:
Also, although this implementation is more concise, my preference would be not to drive the recursion directly by user input. If the splitting would not be limited, the `JAVA_TIME_TO_CHAR_MAPPINGS` would also only need to be walked once (and not restarted on each function call).